### DISCUSSION TITLE
Copilot in VSCode freezes and just shows "Generating..." in the chat.

### DISCUSSION THREAD

**Discussion by jascomp** [community/community#147055] [Copilot Conversations] [answered]
## Select Topic Area
Bug

## Body

Within the last couple of days the Copilot chat has started "freezing" after a few prompts.  The inline code  suggestions also quit working.  Once this happens it will only show "Generating..." (in the case of inline code it just doesn't do anything) and the only way to fix the issue is to restart VSCode.  I'm running my dev environment in WSL.

The host machine:
Version: 1.96.0 (system setup)
Commit: 138f619c86f1199955d53b4166bef66ef252935c
Date: 2024-12-11T02:29:09.626Z
Electron: 32.2.6
ElectronBuildId: 10629634
Chromium: 128.0.6613.186
Node.js: 20.18.1
V8: 12.8.374.38-electron.0
OS: Windows_NT x64 10.0.26100
GitHub Copilot Extension: 1.252.0
GitHub Copilot Chat Extension: 0.23.1

This is eventually logged in the VSCode Output window (same for inline):

`2024-12-13 15:55:12.270 [error] TypeError: fetch failed
    at node:internal/deps/undici/undici:13392:13
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at ek._fetch (/home/jason/.vscode-server/extensions/github.copilot-chat-0.23.1/dist/extension.js:763:5811)
    at Mat (/home/jason/.vscode-server/extensions/github.copilot-chat-0.23.1/dist/extension.js:2212:5824)
    at OO.fetchMany (/home/jason/.vscode-server/extensions/github.copilot-chat-0.23.1/dist/extension.js:2213:1912)
    at CD.detectIntent (/home/jason/.vscode-server/extensions/github.copilot-chat-0.23.1/dist/extension.js:1072:27728)
    at CD.provideParticipantDetection (/home/jason/.vscode-server/extensions/github.copilot-chat-0.23.1/dist/extension.js:1072:26566): Error on conversation request`

**Comment by VenciPython** [comment]
This error means there's an issue with fetch (HTTP network request) in Node.js using the Undici library. It usually occurs due to connectivity issues or a connection timeout. Make sure your VSCode and all extensions are up-to-date. If the problem persists, try disabling and re-enabling the GitHub Copilot extension. If this doesn't work, you can try checking the documentation at https://docs.github.com/en/copilot

**Comment by jascomp** [comment]
Thank you. It does seem to be isolated to WSL. I moved the same Python project over to Windows with no issues (so far). Is there anything WSL/Linux-specific you can advise me to check?

**Reply by VenciPython** [reply, marked as answer]
Make sure that: You are using the latest version of WSL (wsl --update)  and  WSL has proper network connectivity,  and resources are sufficient (CPU, memory),  and   the proxy is configured correctly (if you're using a proxy) :))

**Reply by jascomp** [reply]
@VenciPython  I updated WSL to 2.3.26 (Latest stable): https://github.com/microsoft/WSL/releases.  I did that a couple of hours ago and have been using Copilot through WSL again without an issue.

**Comment by etmoore** [comment]
I'm having the same issue  :/ 

about VSCode:
Version: 1.96.0 (Universal)
Commit: 138f619c86f1199955d53b4166bef66ef252935c
Date: 2024-12-11T02:29:09.626Z
Electron: 32.2.6
ElectronBuildId: 10629634
Chromium: 128.0.6613.186
Node.js: 20.18.1
V8: 12.8.374.38-electron.0
OS: Darwin arm64 23.6.0

**Comment by etmoore** [comment]
opened an issue: https://github.com/microsoft/vscode-copilot-release/issues/3357
